### DISCUSSION TITLE
Apple Home kit take to much time to update devices

### DISCUSSION THREAD

**Discussion by MatthieuCa** [matter-js/matter.js#1155] [Q&A] [answered]
Hello everyone!

I need your help, I have created an app that allow the user to update the endpoint on live the problem I have is that on google home it takes 30sec to update but on Home Kit it takes like 5-10 minutes to update. Is there a reason or is there a solution to force the update?

Thank you

**Comment by Apollon77** [comment]
Ok, sorry but I do not really understand what exactly you mean?  What exactly can a user update? How he does that? What happens when he does that? Which updates you expect? Code? Debug Logs?

**Reply by MatthieuCa** [reply]
Yeah sorry  here is more information:
A user can during the lifecycle of Matter create or update an endpoint using the Bridge.
After this creation or update of the endpoint, the user have to restart the server to apply the changes. (The changes are in a config file)

So when I restart the server what I expect is that like Google Home all the old endpoint are deleted from apple home kit and the new replace them. The problem is that this replacement take to long (5 to 10 minutes).

I don't have any debug logs or Code because this is not really a problem but more likely an optimization of the time apple take to update

**Reply by Apollon77** [reply, marked as answer]
Hm .... Ok ... some comments here ....

matter.js supports "hot editing of endpoints in a bridge" and I would propose you to use this instead of shutting down the device and restarting it completely, because then the changes are synced within 30-60seconds.

What you do means:
* You stop the device. All active sessions are removed from the device. Apple and Google could detect this but they do not do this ... this means they expect a next subscription "update"/"ping" after a time that was agreed about earlier ... this time is ecosystem dependent but normally between 60 and 300s
* You restart the device and the device announce itself ... but the ecosystems still have an active subscription they ignore this
* So in the eyes of the ecosystem the device is still online and sessions are still valid, but the device do not know about this
* Then when subscription time expires (plus x Seconds) then the ecosystem assumes the device being offline and re-established the subscription, potentially initialize new sessions and such

Soooo ... restart the device is kind of the "worst" thing you can do because it exactly leads to the effects you describe :-)

So accept this (at least unless the ecosystems react to shutdown events or such what they ignore right now), or use hot editing of endpoints :-)
